single_include/libndt7.hpp

```diff
@@ -24600,45 +24610,40 @@ Verbosity Client::get_verbosity() const noexcept {
   return settings_.verbosity;
 }
 
-// Function to parse a websocket URL and return its components. The URL must
+const std::regex url_regex(
+      "^([^:/]+)"              // scheme (group 1)
+      "://"                    // constant URI string
+         "("                   // hostname (group 2)
+           "(?:[^/:]*)"        //   standard hostname, may be empty
+           "|"                 //   OR
+           "(?:\\[[^\\]]+\\])" //   everything between two [] brackets, e.g. ipv6.
+         ")"                   //
+      "(?::(\\d+))?"           // port, if present (group 3)
+      "(/.*)?"                 // path and query, if present (group 4)
```

Comment:
```suggestion
      "(/.*)?$"                 // path and query, if present (group 4)
```

We want the regexp to match the whole URL.